File: AndroidDefaultAppBar/01_navigation/NavigationDefaultBottom/app/src/main/java/com/haoyudu/navigationdefaultbottom/MainActivity.java
package com.haoyudu.navigationdefaultbottom;

import android.os.Bundle;
import com.google.android.material.bottomnavigation.BottomNavigationView;
import androidx.appcompat.app.AppCompatActivity;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.navigation.ui.AppBarConfiguration;
import androidx.navigation.ui.NavigationUI;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        BottomNavigationView navView = findViewById(R.id.nav_view);
        //创建一个空的appbar应用组件框架(以MenuView作为参数)
        AppBarConfiguration appBarConfiguration = new AppBarConfiguration.Builder(
                navView.getMenu())
                .build();
        NavController navController = Navigation.findNavController(this, R.id.nav_host_fragment);
        //创建NavigationUI对象
        NavigationUI.setupActionBarWithNavController(this, navController, appBarConfiguration);
        NavigationUI.setupWithNavController(navView, navController);
    }

}

File: AndroidViewPager/ViewPagerTest/app/src/main/java/com/haoyudu/galleryself/MainActivity.kt
package com.haoyudu.galleryself

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.navigation.findNavController
import androidx.navigation.ui.NavigationUI

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        NavigationUI.setupActionBarWithNavController(this,findNavController(R.id.fragment))
    }

    /**重写导航返回事件*/
    override fun onSupportNavigateUp(): Boolean {
        return super.onSupportNavigateUp()||findNavController(R.id.fragment).navigateUp()
    }
}